docs/core/tools/dotnet-environment-variables.md

```diff
@@ -183,6 +183,22 @@ See [EventPipe environment variables](../diagnostics/eventpipe.md#trace-using-en
 
 Specifies the location of the .NET runtimes, if they are not installed in the default location. The default location on Windows is `C:\Program Files\dotnet`. The default location on Linux and macOS is `/usr/local/share/dotnet`. This environment variable is used only when running apps via generated executables (apphosts). `DOTNET_ROOT(x86)` is used instead when running a 32-bit executable on a 64-bit OS.
 
+### `DOTNET_HOST_PATH`
+
+Specifies the absolute path to a `dotnet` binary (`dotnet.exe` on Windows, `dotnet` on non-Windows) that was used to launch this session. This is typically used by a hosting application that wants to ensure that the exact same version of the host binary is used for all child processes.
+
+Without this environment variable, if a `dotnet` binary was used from a location not on the `PATH`, child processes of that `dotnet` that wished to invoke `dotnet` would invoke a different `dotnet` binary than the one that was used to spawn them - this can result in confusing behavior for end users.
+
+As a tool author, if you want to locate the correct `dotnet` binary to use to spawn a .NET binary, you should use the following algorithm:
```

Comment:
Is this only used for child-processes of the launched dotnet? In other words, the initial launch of dotnet is still controlled by the path (when typing dotnet) or the customer (when specifying the full path to dotnet)

Reply:
it's a weird scenario yeah - 

* user invokes `dotnet` binary via absolute path or implicitly by their shell discovering the `dotnet` binary on PATH
* `dotnet` loads SDK
* SDK sets path to the user-selected `dotnet` to `DOTNET_HOST_PATH`
* tools in the SDK that need to execute a standalone `dotnet` command use `DOTNET_HOST_PATH`.

so the value only directly impacts those subprocesses, which I will make clearer.